src/store/modules/app.js

```diff
@@ -73,6 +84,46 @@ const getters = {
 };
 
 const actions = {
+  checkValidKeymap(_, keymap) {
+    if (
+      isUndefined(keymap.keyboard) ||
+      isUndefined(keymap.keymap) ||
+      isUndefined(keymap.layout) ||
+      isUndefined(keymap.layers)
+    ) {
+      return false;
+    }
+    return true;
+  },
+  validateKeyboard({ state }, keyboard) {
+    const valid = state.keyboards.includes(keyboard);
+    console.info(`Validate keyboard:${keyboard} valid:${valid}`);
+    return valid;
+  },
+  async fetchKeyboards({ commit }) {
+    const r = await axios.get(backend_keyboards_url);
+    if (r.status === 200) {
+      const exclude = getExclusionList();
+      commit(
+        'setKeyboards',
+        r.data.filter(keeb => {
+          return isUndefined(exclude[keeb]);
+        })
+      );
+    }
```

Comment:
you should return r

Reply:
i don't agree with it, because when you call fetchKeyboards you don't want to deal with the HTTP response, just the data. Also this is just a void.